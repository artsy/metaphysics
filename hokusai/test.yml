version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
