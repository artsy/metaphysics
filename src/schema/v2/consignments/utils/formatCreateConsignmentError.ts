import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
