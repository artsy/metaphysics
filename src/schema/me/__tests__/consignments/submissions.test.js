import { runAuthenticatedQuery } from "test/utils"
import gql from "test/gql"

describe("submissions", () => {
  it("asks for a user's submissions", () => {
    const mutation = gql`
      {
        me {
          consignment_submissions(first: 5, completed: true) {
            edges {
              node {
                id
                authenticity_certificate
                title
                artist_id
                artist {
                  name
                }
              }
             }
          }
        }
      }
    `

    const rootValue = {
      submissionsLoader: () =>
        Promise.resolve([
          {
            id: "106",
            authenticity_certificate: true,
            artist_id: "123",
            title: "The best photo yet",
          },
        ]),
      artistLoader: () =>
        Promise.resolve({
          name: "Larissa Croft",
          birthday: "April 2011",
          artworks_count: 1,
        }),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(mutation, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
