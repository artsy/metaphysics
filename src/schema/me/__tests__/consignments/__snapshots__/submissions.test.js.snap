// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submissions asks for a user's submissions 1`] = `
Object {
  "me": Object {
    "consignment_submissions": Object {
      "edges": Array [
        Object {
          "node": Object {
            "artist": Object {
              "name": "Larissa Croft",
            },
            "artist_id": "123",
            "authenticity_certificate": true,
            "id": "106",
            "title": "The best photo yet",
          },
        },
      ],
    },
  },
}
`;
